rust/hyperlane-base/src/types/s3_storage.rs

```diff
@@ -220,6 +220,6 @@ impl CheckpointSyncer for S3Storage {
     }
 
     fn announcement_location(&self) -> String {
-        format!("s3://{}/{}", self.bucket, self.region.name())
+        format!("s3://{}/{}/{}", self.bucket, self.region.name(), self.folder)
```

Comment:
What will this show if `self.folder` is `None`?

Reply:
It cannot be None right? I made the `folder` field in S3Storage a String, so worst case it's an empty string, which should technically work